algorithms/dp/OptimalBST.py

```diff
@@ -0,0 +1,54 @@
+'''Construct a binary search tree of all keys such that the total cost of all the searches is as small as possible.'''
+
+######   O(N^3)  ######
+
+from itertools import accumulate
+
+
+def dp(a,acc,n):
+    cost = [[float('inf')]*n for i in range(n)]
+    st = [[0]*n for i in range(n)]
+
+    for i in range(n):
+        cost[i][i] = a[i][1]
+
+    for l in range(2,n+1):
+        for i in range(n-l+1):
+            j = i+l-1
+            for r in range(i,j+1):
+
+                if i!=0:
+                    c =  (lambda r:cost[i][r-1] if r>i else 0)(r) + (lambda r:cost[r+1][j] if r<j else 0)(r) + acc[j] - acc[i-1]
+                else:
+                    c = (lambda r: cost[i][r - 1] if r > i else 0)(r) + (lambda r: cost[r + 1][j] if r < j else 0)(r) + acc[j]
```

Comment:
can we line break to fit this into 80 characters per line? 
ex)

```
c = ((lambda r: cost[i][r - 1] if r > i else 0)(r) +
     (lambda r: cost[r + 1][j] if r < j else 0)(r) +
     acc[j])
```
and can we improve readability (better name for c and r)?